### ISSUE TITLE
"This function should be called while holding treeLock" (MAC OS)

### ISSUE THREAD

**Issue by dfabulich** [bazelbuild/intellij#45]
While testing the IntelliJ plugin today, I got this error on first sync of a new project. I can't seem to reproduce it reliably, but the stacktrace looks useful.

```
This function should be called while holding treeLock
java.lang.IllegalStateException: This function should be called while holding treeLock
	at java.awt.Component.checkTreeLock(Component.java:1201)
	at java.awt.Container.validateTree(Container.java:1689)
	at com.intellij.openapi.actionSystem.impl.ActionMenu.access$900(ActionMenu.java:51)
	at com.intellij.openapi.actionSystem.impl.ActionMenu$MenuItemSynchronizer.propertyChange(ActionMenu.java:334)
	at java.beans.PropertyChangeSupport.fire(PropertyChangeSupport.java:335)
	at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:327)
	at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:263)
	at java.beans.PropertyChangeSupport.firePropertyChange(PropertyChangeSupport.java:303)
	at com.intellij.openapi.actionSystem.Presentation.firePropertyChange(Presentation.java:297)
	at com.intellij.openapi.actionSystem.Presentation.setVisible(Presentation.java:265)
	at com.intellij.openapi.actionSystem.Presentation.setEnabledAndVisible(Presentation.java:292)
	at com.google.idea.blaze.base.actions.BlazeMenuGroup.update(BlazeMenuGroup.java:29)
	at com.intellij.openapi.wm.impl.IdeMenuBar.expandActionGroup(IdeMenuBar.java:428)
	at com.intellij.openapi.wm.impl.IdeMenuBar.updateMenuActions(IdeMenuBar.java:338)
	at com.intellij.openapi.wm.impl.IdeMenuBar$MyTimerListener.run(IdeMenuBar.java:488)
	at com.intellij.openapi.actionSystem.impl.WeakTimerListener.run(WeakTimerListener.java:54)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.runListenerAction(ActionManagerImpl.java:1414)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.notifyListeners(ActionManagerImpl.java:1404)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.actionPerformed(ActionManagerImpl.java:1394)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by brendandouglas** [comment]
Thanks for letting us know. This looks to be specific to macs, which explains why we haven't yet seen it.

I'll leave this open, for when we do eventually get around to adding mac support. In the meantime, it looks relatively benign.

**Comment by drichelson** [comment]
Fyi I'm seeing this in CLion on Mac OS:
```CLion 2018.2.6
Build #CL-182.5107.21, built on November 12, 2018
...
JRE: 1.8.0_152-release-1248-b22 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.6```

**Comment by chaoren** [comment]
This looks the same as https://github.com/bazelbuild/intellij/issues/200.
Should be fixed here: https://github.com/JetBrains/intellij-community/pull/705.

**Comment by chaoren** [comment]
The fix is in 2018.3.
